pkg/provision/providers/qemu/tpm2.go

```diff
@@ -0,0 +1,56 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package qemu
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/hashicorp/go-multierror"
+
+	"github.com/siderolabs/talos/pkg/provision"
+	"github.com/siderolabs/talos/pkg/provision/providers/vm"
+)
+
+func (p *provisioner) createVirtualTPM2State(state *vm.State, nodeName string) (tpm2Config, error) {
+	tpm2StateDir := state.GetRelativePath(fmt.Sprintf("%s-tpm2", nodeName))
+
+	if err := os.MkdirAll(tpm2StateDir, 0o755); err != nil {
+		return tpm2Config{}, err
+	}
+
+	return tpm2Config{
+		NodeName: nodeName,
+		StateDir: tpm2StateDir,
+	}, nil
+}
+
+func (p *provisioner) destroyVirtualTPM2s(cluster provision.ClusterInfo) error {
+	errCh := make(chan error)
+
+	nodes := append([]provision.NodeInfo{}, cluster.Nodes...)
+
+	for _, node := range nodes {
+		tpm2PidPath := filepath.Join(node.TPM2StateDir, "swtpm.pid")
```

Comment:
what if secure boot is not enabled? should we check for `TPM2StateDir` being set?

Reply:
tpm2 is independent of secureboot, `TPM2StateDir` is set when `--with-tpm2` is passed

Reply:
I mean this code runs for `cluster destroy`, so if `--with-secureboot` is not set, this code will try to join empty `node.TPM2StateDir`

Reply:
not sure I get that, `node.TPM2StateDir` is set only when `--with-tpm2` is passed

Reply:
oh, I see that now, got it